File: podklady/6_Others/2_github_copilot_sdk/typescript/src/1_single_agent/0a_simplest_agent.ts
/**
 * 0a — The simplest possible Copilot SDK session.
 *
 * Concept
 * -------
 * Spin up a CopilotClient, create a session, send one prompt, print the answer,
 * shut down. No tools, no streaming, no custom config.
 *
 * Run:
 *   npx tsx src/1_single_agent/0a_simplest_agent.ts
 */

import { approveAll, CopilotClient } from "@github/copilot-sdk";

async function main(): Promise<void> {
	const client = new CopilotClient();

	// Sessions are the unit of conversation. They are stateful by default,
	// which is the equivalent of `ClaudeSDKClient` in the Claude Agent SDK.
	const session = await client.createSession({
		onPermissionRequest: approveAll,
	});

	const reply = await session.sendAndWait(
		{ prompt: "What is 2 + 2?" },
		300_000,
	);

	if (reply) {
		// The final assistant message lives on `reply.data.content`.
		console.log((reply.data as { content: string }).content);
	}

	await session.disconnect();
	await client.stop();
}

main().catch(console.error);

File: podklady/6_Others/2_github_copilot_sdk/python/1_single_agent/0a_simplest_agent.py
"""
0a — The simplest possible Copilot SDK session.

Concept
-------
Spin up a CopilotClient, create a session, send one prompt, print the answer,
shut down. No tools, no streaming, no custom config.

Run:
    python 1_single_agent/0a_simplest_agent.py
"""

import asyncio

from copilot import CopilotClient
from copilot.session import PermissionHandler


async def main() -> None:
    client = CopilotClient()
    await client.start()

    # Sessions are the unit of conversation. They are stateful by default,
    # which is the equivalent of `ClaudeSDKClient` in the Claude Agent SDK.
    session = await client.create_session(
        on_permission_request=PermissionHandler.approve_all,
    )

    reply = await session.send_and_wait("What is 2 + 2?")

    if reply is not None:
        # The final assistant message lives on `reply.data.content`.
        print(reply.data.content)

    await session.disconnect()
    await client.stop()


if __name__ == "__main__":
    asyncio.run(main())
